src/txmempool.cpp

```diff
@@ -1043,17 +1043,23 @@ void CTxMemPool::TrimToSize(size_t sizelimit, std::vector<COutPoint>* pvNoSpends
 
     unsigned nTxnRemoved = 0;
     CFeeRate maxFeeRateRemoved(0);
-    while (!mapTx.empty() && DynamicMemoryUsage() > sizelimit) {
+    while (!mapTx.empty()) {
         indexed_transaction_set::index<descendant_score>::type::iterator it = mapTx.get<descendant_score>().begin();
 
+        CFeeRate removed(it->GetModFeesWithDescendants(), it->GetSizeWithDescendants());
```

Comment:
I think it would be good to rename this -- previously the tx was definitely being removed, but now the removal is conditional. `candidate` maybe?